.github/workflows/test.yml

```
@@ -79,27 +79,25 @@ jobs:
           restore-keys: ${{ runner.os }}-build-${{ hashFiles('**/yarn.lock') }}-
       - run: yarn prepare
       - run: yarn build
-      - uses: actions/upload-artifact@v3
-        with:
-          name: build
-          path: build
-          if-no-files-found: error
+      - run: yarn test:size
 
-  size-tests:
-    needs: [build]
+  build-e2e:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/setup
```

caching will speed this up by 2x.

```suggestion
      - uses: ./.github/actions/setup
      - uses: actions/cache@v3
        id: build-e2e-cache
        with:
          path: node_modules/.cache
          key: ${{ runner.os }}-build-e2e-${{ hashFiles('**/yarn.lock') }}-${{ github.run_id }}
```